#!/usr/bin/env python
PACKAGE = "pointcloud_to_ocgrid"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# variables
gen.add("ocgridWidth",   int_t,   0, "The width (cells) of the ocgrid", 400, 1, 1000)
gen.add("ocgridHeight",   int_t,   0, "The height (cells) of the ocgrid", 200, 1, 1000)
gen.add("ocgridResolution",   double_t,   0, "The resolution (m/cell) of the ocgrid", 0.02, 0.001, 1.0)
gen.add("ocgridXOrigin",   double_t,   0, "The X origin of the bottom left cell", -1.0, -10.0, 10.0)
gen.add("ocgridYOrigin",   double_t,   0, "The Y origin of the bottom left cell", -1.0, -10.0, 10.0)

gen.add("pointcloudZMin",   double_t,   0, "The lower bound of pointcloud z filtering", -0.2, -1.0, 1.0)
gen.add("pointcloudZMax",   double_t,   0, "The upper bound of pointcloud z filtering", 0.1, -1.0, 1.0)
gen.add("inflation",   int_t,   0, "Inflation for occupied cells", 2, 0, 10)

exit(gen.generate(PACKAGE, "pointcloud_to_ocgrid", "settings"))